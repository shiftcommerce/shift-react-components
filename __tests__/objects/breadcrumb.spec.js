// Libraries
import React from 'react'

// Objects
import Breadcrumb from '../../src/objects/breadcrumb'

// Fixtures
import breadcrumbMenuTrail from '../fixtures/breadcrumbs'

test('renders correctly', () => {
  // Arrange & Act
  const wrapper = mount(
    <Breadcrumb trail={breadcrumbMenuTrail} />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toContainMatchingElements(3, 'a.o-breadcrumb__link')
})
