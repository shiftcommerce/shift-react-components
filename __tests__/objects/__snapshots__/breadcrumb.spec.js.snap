// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Breadcrumb
  trail={
    Array [
      Object {
        "canonical_path": "/categories/22bca338-b578-4aee-a302-992764fa3e72",
        "id": "22bca338-b578-4aee-a302-992764fa3e72",
        "page": "/categories/category",
        "title": "Womens",
      },
      Object {
        "canonical_path": "/products/708a0142-e60f-4332-9a8b-1359c5af9ec4",
        "id": "708a0142-e60f-4332-9a8b-1359c5af9ec4",
        "page": "/products/product",
        "title": "Textured Long T-Shirt",
      },
    ]
  }
>
  <nav>
    <ol
      className="o-breadcrumb"
    >
      <li
        className="o-breadcrumb__item"
      >
        <Link
          as="/"
          className="o-breadcrumb__link"
          href="/home/index"
        >
          <a
            className="o-breadcrumb__link"
            href="/home/index"
          >
            Home
          </a>
        </Link>
      </li>
      <li
        className="o-breadcrumb__item"
        key="0"
      >
        <Link
          as="/categories/22bca338-b578-4aee-a302-992764fa3e72"
          className="o-breadcrumb__link"
          href="/categories/category?id=22bca338-b578-4aee-a302-992764fa3e72"
        >
          <a
            className="o-breadcrumb__link"
            href="/categories/category?id=22bca338-b578-4aee-a302-992764fa3e72"
          >
            Womens
          </a>
        </Link>
      </li>
      <li
        className="o-breadcrumb__item"
        key="1"
      >
        <Link
          as="/products/708a0142-e60f-4332-9a8b-1359c5af9ec4"
          className="o-breadcrumb__link"
          href="/products/product?id=708a0142-e60f-4332-9a8b-1359c5af9ec4"
        >
          <a
            className="o-breadcrumb__link"
            href="/products/product?id=708a0142-e60f-4332-9a8b-1359c5af9ec4"
          >
            Textured Long T-Shirt
          </a>
        </Link>
      </li>
    </ol>
  </nav>
</Breadcrumb>
`;
